File: 933-NumberofRecentCalls/933-NumberofRecentCalls.cpp
// Last updated: 6/22/2026, 11:03:29 AM
1class Solution {
2public:
3    ListNode* deleteMiddle(ListNode* head) {
4        if (head == nullptr || head->next == nullptr)
5            return nullptr;
6
7        int n = 0;
8        ListNode* temp = head;
9
10        while (temp) {
11            n++;
12            temp = temp->next;
13        }
14
15        int mid = n / 2;
16
17        temp = head;
18        for (int i = 1; i < mid; i++) {
19            temp = temp->next;
20        }
21
22        temp->next = temp->next->next;
23
24        return head;
25    }
26};

File: 933-NumberofRecentCalls/933-NumberofRecentCalls.java
// Last updated: 6/28/2026, 7:39:47 AM
1class Solution {
2    public String predictPartyVictory(String senate) {
3
4        int n = senate.length();
5
6        Queue<Integer> radiant = new LinkedList<>();
7        Queue<Integer> dire = new LinkedList<>();
8
9        for (int i = 0; i < n; i++) {
10            if (senate.charAt(i) == 'R')
11                radiant.add(i);
12            else
13                dire.add(i);
14        }
15
16        while (!radiant.isEmpty() && !dire.isEmpty()) {
17
18            int r = radiant.poll();
19            int d = dire.poll();
20
21            if (r < d) {
22                radiant.add(r + n);
23            } else {
24                dire.add(d + n);
25            }
26        }
27
28        return radiant.isEmpty() ? "Dire" : "Radiant";
29    }
30}